src/main/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStepExecution.java

```diff
@@ -281,80 +136,54 @@ private static void println(StepContext context, String message) {
         try {
             context.get(TaskListener.class).getLogger().println(message);
         } catch (Exception x) {
-            LOGGER.log(WARNING, "failed to print message to dead " + context, x);
+            LOGGER.log(Level.WARNING, x, () -> "failed to print message to dead " + context);
         }
     }
 
-    private static void cancel(StepContext context, Integer build) throws IOException, InterruptedException {
-        if (context.isReady()) {
-            println(context, "Canceled since build #" + build + " already got here");
```

Comment:
AFAICT this has been lost?

Reply:
No longer needed with the current algorithm